src/traits/networking/centralized_server_network.rs

```diff
@@ -1021,6 +1023,32 @@ where
     }
 }
 
+impl<
+        TYPES: NodeType,
+        I: NodeImplementation<TYPES>,
+        PROPOSAL: ProposalType<NodeType = TYPES>,
+        VOTE: VoteType<TYPES>,
+        MEMBERSHIP: Membership<TYPES>,
+    > TestableNetworkingImplementation<TYPES, Message<TYPES, I>>
+    for CentralizedCommChannel<TYPES, I, PROPOSAL, VOTE, MEMBERSHIP>
+where
+    TYPES::SignatureKey: TestableSignatureKey,
+    MessageKind<TYPES::ConsensusType, TYPES, I>: ViewMessage<TYPES>,
+{
+    fn generator(
+        expected_node_count: usize,
+        num_bootstrap: usize,
+        network_id: usize,
+    ) -> Box<dyn Fn(u64) -> Self + 'static> {
+        let generator=<CentralizedServerNetwork<TYPES::SignatureKey, TYPES::ElectionConfigType> as TestableNetworkingImplementation<TYPES, Message<TYPES, I>>>::generator(expected_node_count, num_bootstrap, network_id);
```

Comment:
I'm surprised cargo fmt was okay with no spaces between `generator` and `<`. Might be worth trying to format only this file.

Reply:
Good catch! Auto-formating doesn't seem to work for complicated types like this. I'll manually fix this one. I've manually formatted some other places but missed this one. 😅